test/entrypoint.sh

```diff
@@ -12,10 +12,19 @@ while ! exec 6<>/dev/tcp/0.0.0.0/3306; do
     sleep 1
 done
 
+# make sure we can reach the rabbitmq
+while ! exec 6<>/dev/tcp/0.0.0.0/5672; do
```

Comment:
Good questions. The `exec` commands will open the sockets and assign to the file descriptor. Note it neither reads nor writes. It should work since we are not interested in reading/writing from the file descriptors, but only opening the socket by assigning it to a file descriptor.

According to this link the file descriptor is exhausted when it is read: http://www.catonmat.net/blog/bash-one-liners-explained-part-three/

Although I'm not an UNIX expert I can't see how it could make any harm besides not working, since the file descriptor is closed for both reading and writing (I assume) in the lines 21-22.

You can test it by commenting out the rabbitmq container execution from `test/run-docker.sh`.

Do you think it's better to do the entire create file descriptor, close file descriptor for reading and writing dance for this too?